vignettes/fims-setup-guide.Rmd

```diff
@@ -0,0 +1,177 @@
+---
+title: "FIMS User Setup Guide"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{FIMS User Setup Guide}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+```{r, functions, include=FALSE}
+# Helper function to extract remote markdown from FIMS .devcontainer/ README to reuse the materials
+extract_remote_markdown <- function(url, header_pattern) {
+  lines <- try(readLines(url, warn = FALSE), silent = TRUE)
+  tmp <- tempfile(fileext = ".md")
+  
+  if (inherits(lines, "try-error")) {
+    writeLines("_Documentation temporarily unreachable._", tmp)
+    return(tmp)
+  }
+  
+  # Find the starting line (e.g., the line containing "## WSL2")
+  start_idx <- grep(header_pattern, lines, ignore.case = TRUE)
+  
+  if (length(start_idx) > 0) {
+    # Determine the level of the header found (count the # symbols)
+    header_line <- lines[start_idx]
+    header_level <- nchar(gsub("(^#+).*", "\\1", header_line))
+    
+    # Find headers that are at the SAME level or HIGHER (fewer or equal #)
+    # We want to ignore ### if we are looking for a ## section
+    stop_pattern <- paste0("^#{1,", header_level, "} ")
+    all_potential_stops <- grep(stop_pattern, lines)
+    
+    # The end index is the next header that isn't the current one
+    end_idx <- all_potential_stops[all_potential_stops > start_idx][1]
+    
+    if (is.na(end_idx)) end_idx <- length(lines) else end_idx <- end_idx - 1
+    
+    # Extract and add blank lines for proper RMarkdown list rendering
+    content <- c("", lines[(start_idx + 1):end_idx], "") 
+    writeLines(content, tmp)
+  } else {
+    writeLines("_Section not found in remote documentation._", tmp)
+  }
+  
+  return(tmp)
+}
+```
+
+## Overview
+
+This guide facilitates the configuration of a standardized Fisheries Integrated Modeling System (FIMS) environment. By utilizing cloud-based workstations and containerized development environments, we ensure cross-platform consistency and eliminate the complexities of manual dependency management.
```

Comment:
Do we need explainers on some of these technical concepts? I feel like some people may be unfamiliar with the terms "containerized development environment" "cross-platform consistency" and "dependency management". We could either explain the terms or we could try to use less technical language.